lib/static/modules/actions/static-accepter.ts

```diff
@@ -58,10 +63,13 @@ export const staticAccepterCommitScreenshot = (
         axiosRequestOptions = {},
         meta
     }: StaticAccepterCommitScreenshotOptions
-): ThunkAction<Promise<void>, Store, void, StaticAccepterCommitScreenshotAction> => {
-    return async (dispatch: Dispatch) => {
+): ThunkAction<Promise<CommitResult>, Store, void, StaticAccepterCommitScreenshotAction> => {
+    return async (dispatch: Dispatch): Promise<CommitResult> => {
         dispatch({type: actionNames.PROCESS_BEGIN});
         dispatch(staticAccepterCloseConfirm());
+        dispatch({type: actionNames.UPDATE_LOADING_IS_IN_PROGRESS, payload: true});
+        dispatch({type: actionNames.UPDATE_LOADING_TITLE, payload: `Preparing images to commit: 0 of ${imagesInfo.length}`});
```

Comment:
I've thought about the idea where reducers can automatically determine whether progress bar should be shown. In fact, it was like that before. However, there are too many edge cases that make this too complicated and I decided to stick to a more low-level API. This include:

- Showing/hiding progress bar on app load. It requires very specific timings, for example we add 500ms delay before showing progress bar so that if app loads instantly, the progress bar won't blink. We also add a slight delay when hiding it, because otherwise it will lag as too many items re-render at once

- We need a delay when showing message at the end of a task, e.g. "Images uploaded!"

These and other peculiarities are difficult to implement on the reducers' end for "automatic" handling of loading. But they are important for end user experience.

I suggest to keep it as is for now and if we stumble upon cases where overriding progress message is an issue, deal with it at that point.